test/views/html/taxCodeChange/TaxCodeComparisonViewSpec.scala

```diff
@@ -81,26 +81,59 @@ class TaxCodeComparisonViewSpec extends TaiViewSpec {
     }
 
     "display a button linking to the 'check your tax-free amount page" in {
-      doc(view) must haveLinkElement("check-your-tax-button", routes.TaxCodeChangeController.yourTaxFreeAmount().url.toString, Messages("taxCode.change.yourTaxCodeChanged.checkYourTaxButton"))
+      doc must haveLinkElement("" +
+        "check-your-tax-button",
+        routes.TaxCodeChangeController.yourTaxFreeAmount().url.toString,
+        Messages("taxCode.change.yourTaxCodeChanged.checkYourTaxButton")
+      )
     }
 
     "display the pension number" when {
       "a pension" in {
         val expectedText = Messages("tai.pensionNumber") + ": 1234 " + Messages("tai.pension.income.details.pensionNumber.screenReader", "1234")
-        doc(view) must haveClassWithText(expectedText, "tax-code-change__payroll")
+        doc must haveClassWithText(expectedText, "tax-code-change__payroll")
       }
     }
 
     "display the employment number" when {
       "a employment" in {
         val expectedText = Messages("tai.payRollNumber") + ": Payroll Number " + Messages("tai.employment.income.details.payrollNumber.screenReader", "Payroll Number")
-        doc(view) must haveClassWithText(expectedText, "tax-code-change__payroll")
+        doc must haveClassWithText(expectedText, "tax-code-change__payroll")
       }
     }
 
     "does not display payroll or pension number when the payrollNumber does not exist" in {
       doc must haveClassCount("tax-code-change__payroll", 4)
       doc must haveClassCount("tax-code-change__payroll-number", 3)
     }
+
+    "display tax code change reasons" when {
+      def createTaxCodeRecord(employerName: String): TaxCodeRecord = {
+        TaxCodeRecord("1185L", startDate, startDate.plusMonths(1), OtherBasisOfOperation, employerName, true, Some("1234"), true)
+      }
+
+      "primary employments have changed" in {
+        val previous = createTaxCodeRecord("Employer 1")
+        val current = createTaxCodeRecord("Employer 2")
+        val taxCodeChange: TaxCodeChange = TaxCodeChange(Seq(previous), Seq(current))
+        val viewModel: TaxCodeChangeViewModel = TaxCodeChangeViewModel(taxCodeChange, Map[String, BigDecimal]())
+
+        val view = views.html.taxCodeChange.taxCodeComparison(viewModel)
+        doc(view) must haveClassCount("tax-code-reason", 2)
+      }
+
+      "display a generic tax code reason" when {
+        "there are more than 4 reasons" in {
+          val fivePreviousJobs: Seq[TaxCodeRecord] = Seq.tabulate(5)(i => createTaxCodeRecord(s"Employer+$i"))
```

Comment:
Interesting way of creating the five employments. I assume for this test we are simply bothered about the number of records in the seq?